### PR TITLE
Add `DO` nesting edge cases

### PR DESCRIPTION
**Pull request by konradha**
The fixed form tokenizer will need to understand more edge cases in loops.

### CODE REVIEW

**Comment by konradha** [comment]
When we have _most_ edge cases for `DO` and `IF` related to `CONTINUE`, we can enable lexing `continue` in the body -- that enables us to tokenize the largest body of code in SciPy.

File: tests/loop_test2.f [hunk 1]
```diff
@@ -0,0 +1,37 @@
+c `--show-llvm` throws with
+c assert(Ty && "Invalid GetElementPtrInst indices for type!"); 
+c 
+c -> correct as we do not define lower/upper bounds for arrays here
```
Comment (certik) [inline tests/loop_test2.f:4]: LLVM should never throw any such exceptions, we should give an error message in AST->ASR phase, or LLVM should say "not implemented yet". Also our verify() pass should catch everything invalid.

**Review by certik — APPROVED** [review]
This is fine. We can later refine it as well.
